torchvision/models/detection/rpn.py

```diff
@@ -354,7 +354,17 @@ def _get_top_n_idx(self, objectness, num_anchors_per_level):
         # type: (Tensor, List[int])
         r = []
         offset = 0
-        for ob in objectness.split(num_anchors_per_level, 1):
+        if torchvision._is_tracing():
```

Comment:
It now looks like this function is completely different (or almost) for ONNX and the other code-path.

Also, would this become unnecessary when https://github.com/pytorch/pytorch/pull/32493 gets merged?

Reply:
The PR is not addressing this issue. Currently split with dynamic sizes is not supported in tracing.
I agree  that the behavior is much different here. I can implement _get_top_n_idx_for_onnx.
What do you think?

Reply:
I'm actually going to add support for split with dynamic input in the exporter. We won't need these extra code-paths anymore.